### PR TITLE
Add missing up-to-date configuration

### LINKED ISSUE #35204: [CI] JarHellTaskIT Jar Hell not detected
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+release-tests/50/console

Reproduces with 
```
./gradlew :build-tools:integTest
```

```
org.elasticsearch.gradle.precommit.JarHellTaskIT > testJarHellDetected STANDARD_ERROR
    SLF4J: Class path contains multiple SLF4J bindings.
    SLF4J: Found binding in [jar:file:/var/lib/jenkins/.gradle/caches/4.10/generated-gradle-jars/gradle-api-4.10.jar!/org/slf4j/impl/StaticLoggerBinder.class]
    SLF4J: Found binding in [jar:file:/var/lib/jenkins/.gradle/caches/modules-2/files-2.1/org.gradle/gradle-logging/4.3/d2e47d6a87e11fceddaeb08bbcf7fe1e79e10621/gradle-logging-4.3.jar!/org/slf4j/impl/StaticLoggerBinder.class]
    SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
    SLF4J: Actual binding is of type [org.gradle.internal.logging.slf4j.OutputEventListenerBackedLoggerContext]

org.elasticsearch.gradle.precommit.JarHellTaskIT > testJarHellDetected FAILED
    java.lang.AssertionError: Expected the following line in output:

    java.lang.IllegalStateException: jar hell!
```

There is a second error in the log, I'm not clear if this is the cause of the first:

```
* What went wrong:
Could not resolve all files for configuration ':loggerUsagePlugin'.
> Could not find org.elasticsearch.test:logger-usage:7.0.0-alpha1.
  Searched in the following locations:
    - https://repo.maven.apache.org/maven2/org/elasticsearch/test/logger-usage/7.0.0-alpha1/logger-usage-7.0.0-alpha1.pom
    - https://repo.maven.apache.org/maven2/org/elasticsearch/test/logger-usage/7.0.0-alpha1/logger-usage-7.0.0-alpha1.jar
    - https://artifacts.elastic.co/maven/org/elasticsearch/test/logger-usage/7.0.0-alpha1/logger-usage-7.0.0-alpha1.pom
    - https://artifacts.elastic.co/maven/org/elasticsearch/test/logger-usage/7.0.0-alpha1/logger-usage-7.0.0-alpha1.jar
    - http://s3.amazonaws.com/download.elasticsearch.org/lucenesnapshots/7d0a7782fa/org/elasticsearch/test/logger-usage/7.0.0-alpha1/logger-usage-7.0.0-alpha1.pom
    - http://s3.amazonaws.com/download.elasticsearch.org/lucenesnapshots/7d0a7782fa/org/elasticsearch/test/logger-usage/7.0.0-alpha1/logger-usage-7.0.0-alpha1.jar
    - file:/var/lib/jenkins/workspace/elastic+elasticsearch+master+release-tests/build/local-test-repo/org/elasticsearch/test/logger-usage/7.0.0-alpha1/logger-usage-7.0.0-alpha1.pom
    - file:/var/lib/jenkins/workspace/elastic+elasticsearch+master+release-tests/build/local-test-repo/org/elasticsearch/test/logger-usage/7.0.0-alpha1/logger-usage-7.0.0-alpha1.jar
  Required by:
      project 
```

### PR DESCRIPTION
**Pull request by alpar-t**
The source properties file and the dynamic elasticsearch version
(set based on properties ) were missing from task outputs leading
to the task being incorrectly considered up to date.

Closes #35204

### CODE REVIEW

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -47,6 +47,8 @@ if (project == rootProject) {
 Properties props = VersionPropertiesLoader.loadBuildSrcVersion(project.file('version.properties'))
 version = props.getProperty("elasticsearch")
 processResources {
+  inputs.file(project.file('version.properties'))
+  inputs.property("dynamic_elasticsearch_version", props.getProperty("elasticsearch"))
```
Comment (nik9000) [inline buildSrc/build.gradle]: Do we need this one? The ES version already comes from the file, right?

File: buildSrc/build.gradle [hunk 2]
```diff
@@ -47,6 +47,8 @@ if (project == rootProject) {
 Properties props = VersionPropertiesLoader.loadBuildSrcVersion(project.file('version.properties'))
```
Comment (nik9000) [inline buildSrc/build.gradle]: Maybe make a `File versionProperties = project.file('version.properties')` variable so it is obvious we reference it twice.

File: buildSrc/build.gradle (hunk 1 above)
Comment (atorok) [reply buildSrc/build.gradle]: Yes, we add snapshot and qualifier from parameters to it.

**Comment by atorok** [comment]
@nik9000 thanks for reviewing. I implemented your suggestions and am going to merge because it's causing failures for people locally as the Lucene snapshot was updated.

**Pull request merged**